unlock-app/src/components/interface/locks/List/elements/LockCard.tsx

```diff
@@ -140,18 +175,27 @@ const LockIcon = ({ lock }: LockIconProps) => {
   )
 }
 
-export const LockCard = ({ lock, network, isLoading }: LockCardProps) => {
+export const LockCard = ({ lock, network }: LockCardProps) => {
   const { networks } = useConfig()
+  const web3service = useWeb3Service()
   const lockAddress = lock.address
   const [isCopied, setCopied] = useClipboard(lockAddress, {
     successDuration: 2000,
   })
-
   const { explorer, baseCurrencySymbol } = networks?.[network] ?? {}
 
   const explorerUrl = explorer?.urls?.address(lockAddress) || '#'
 
-  const symbol = (lock as any)?.currencySymbol ?? baseCurrencySymbol
+  const getLockDetail = async () => {
+    return await web3service.getLock(lock.address, parseInt(network, 10))
```

Comment:
No , if you do this it's going to be exactly like what we do now! Please just get the balance... not the whole lock object.